pennylane_cirq/simulator_device.py

```diff
@@ -208,12 +211,60 @@ def generate_samples(self):
 
         self._result = self._simulator.run(self.circuit, repetitions=self.shots)
 
-        # Bring measurements to a more managable form, but keep True/False as values for now
-        # They will be changed in the measurement routines where the observable is available
         return np.array(
             [self._result.measurements[str(wire)].flatten() for wire in range(self.num_wires)]
         ).T.astype(int)
 
+    def expval(self, observable, shot_range=None, bin_size=None):
+        # pylint: disable=missing-function-docstring
+        # Analytic mode
+        if self.shots is None:
+            # Observables are in tensor form
+            if isinstance(observable, qml.operation.Tensor):
+                # Projector, Hamiltonian, Hermitian
+                for name in observable.name:
+                    if self._observable_map[name] is None or name == "Projector":
+                        return super().expval(observable, shot_range, bin_size)
+
+                if "Hadamard" in observable.name:
+                    T = qml.operation.Tensor()
+                    for obs in observable.obs:
+                        T = qml.operation.Tensor(T, qml.PauliZ(wires=obs.wires))
+
+                    circuit = self.circuit
+                    obs = cirq.PauliSum() + self.to_paulistring(T)
```

Comment:
Would there be a way to represent Hadamard using Cirq's `simulate_expectation_values`?

Reply:
No it is not considered as an observable, then we have to bypass it.